### ISSUE TITLE
DnsPlugin Issue

### ISSUE THREAD

**Issue by Kitsune-Akki** [zchoate/LetsEncrypt_Az_AADDS_Renewal#2]
**Fewer DnsPlugin values than names in the order. Using Cloudflare for the rest.**
And looking at the JSON after the job failed:

```
{   **REDACTED**
    "type": "Microsoft.Automation/AutomationAccounts/Jobs",
    "properties": {
        "jobId": "**REDACTED**",
        "creationTime": "2021-03-10T16:19:31.2955324+00:00",
        "provisioningState": "Failed",
        "status": "Failed",
        "statusDetails": "None",
        "startedBy": "{scrubbed}",
        "startTime": "2021-03-10T16:22:16.3193816+00:00",
        "endTime": "2021-03-10T16:22:16.3193816+00:00",
        "lastModifiedTime": "2021-03-10T16:22:16.3193816+00:00",
        "lastStatusModifiedTime": "2021-03-10T16:22:16.3193816+00:00",
        "exception": "The remote server returned an error: (400) Bad Request. (The remote server returned an error: (400) Bad Request.)",
        "parameters": {},
        "runOn": null,
        "runbook": {
            "name": "Install-LE-AADDS"
        }
    }
```
All details are correct - as we tested them (API, etc with another service)

**Comment by zchoate** [comment]
Thanks! I'll take a further look and see what I can find. The error returned is from the POSH-ACME module. Can you confirm the following:
- Using Cloudflare
- dnsApiId is set to the email address on the Cloudflare account
- dnsApiSecret is set to the global API token (not a generated token, I haven't thoroughly tested with that)
- RunAs account is setup and has permission to modify the AADDS resource

**Comment by Kitsune-Akki** [comment]
Hi @zchoate,

We are confirming the info :)

So the dnsApiId uses the email address for the account rather than the API ID itself?

Many thanks for the prompt responses and assistance!

Kind Regards
PS: I will switch over to a work account for future liaison etc.

**Comment by wayne-jfl** [comment]
Hi @zchoate,

So we confirmed the info, the script came back as completed - however new errors:

`You cannot call a method on a null-valued expression. At line:167 char:1 + $cacheItem = $cache.ReadItems() + ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ + CategoryInfo : InvalidOperation: (:) [], RuntimeException + FullyQualifiedErrorId : InvokeMethodOnNull`

`Cannot index into a null array. At line:168 char:1 + $accessToken=($cacheItem | Where-Object { $_.Resource -eq "https://ma ... + ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ + CategoryInfo : InvalidOperation: (:) [], RuntimeException + FullyQualifiedErrorId : NullArray`

`Invoke-RestMethod : {"error":{"code":"AuthenticationFailedMissingToken","message":"Authentication failed. The 'Authorization' header is missing the access token."}} At line:173 char:1 + Invoke-RestMethod -Uri $url -Headers $headerParams -Method Patch -Con ... + ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ + CategoryInfo : InvalidOperation: (System.Net.HttpWebRequest:HttpWebRequest) [Invoke-RestMethod], WebException + FullyQualifiedErrorId : WebCmdletWebResponseException,Microsoft.PowerShell.Commands.InvokeRestMethodCommand`

Not sure if this is related to your script or the POSH-ACME?

Any insights you have are greatly appreciated!

Wayne

**Comment by zchoate** [comment]
Closing as this seems to be addressed in #3.
